bigtable/beam/workload-generator/src/main/java/bigtable/WorkloadGenerator.java

```diff
@@ -88,19 +81,36 @@ private static void cancelJob(BigtableWorkloadOptions options, DataflowPipelineJ
     client.updateJob(jobId, job);
   }
 
-  public static class ReadFromTableFn extends AbstractCloudBigtableTableDoFn<Long, Void> {
+  public static class ReadFromTableFn extends DoFn<Long, Void> {
+
+    private final String projectId;
+    private final String instanceId;
+    private BigtableDataClient bigtableDataClient;
+
+    public ReadFromTableFn(String projectId, String instanceId) {
+      this.projectId = projectId;
+      this.instanceId = instanceId;
+    }
+
+    @Setup
+    public void setup() throws IOException {
+      this.bigtableDataClient = BigtableDataClient.create(this.projectId, this.instanceId);
```

Comment:
Please share the bigtable client across DoFn instances:
```java
class ReadFromTableFn {
   static Object lock = new Object();
   static BigtableDataClient client;
   static int refCount = 0;

   @Setup
   void setup() {
      synchronized(lock) {
          if (++refCount == 1) {
             client = BigtableDataClient.create(...)
          }
      }
   }
   @Teardown
   void teardown() {
      synchronized(lock) {
       if (--refCount == 0) {
          client.close();
          client = null;
       }
     }
   }
}
```

Reply:
Thank you for this code snippet. Updated